docs/webiny-apps/page-builder/development/creating-iframe-element-plugin.md

```diff
@@ -0,0 +1,197 @@
+            },
+            render(props) {
+                return <IFrameEmbed {...props} />;
+            },
+            renderElementPreview({ width, height }) {
+                return <img style={{ width, height }} alt={"iFrame"} />;
+            }
+        } as PbEditorPageElementPlugin,
+    // Settings dialog comes here
+```
+
+The key properties of the plugin are the `create`, `data`, `settings`, and `render`. They contribute to the state of our plugin. On changing the state, it’s up to the render function to render the component based on the state changes.
+
+Since we are adding a new element in the editor, the type of plugin will be `pb-editor-page-element`. This plugin consists of different properties, one of them is the `toolbar` property that helps us put our element on the media group, using `pb-editor-element-group-media`, as seen in the image below.
+
+![Editor Iframe Element](/img/webiny-apps/page-builder/development/development/plugin-reference/editor/iframe/editor-iframe-plugin.png)
+
+### Render Plugin
+
+The next plugin type we will use is [`pb-editor-page-element-advanced-settings`](/docs/webiny-apps/page-builder/development/plugins-reference/app#pb-editor-page-element-advanced-settings)
+, we will use this plugin to show a settings dialog so we can provide an iframe URL.
+
+```jsx
+import { Tab } from "@webiny/ui/Tabs";
+import { Input } from "@webiny/ui/Input";
+import { Grid, Cell } from "@webiny/ui/Grid";
+import {
+    PbEditorPageElementAdvancedSettingsPlugin
+} from "@webiny/app-page-builder/types";
+import { validation } from "@webiny/validation";
+
+{
+    name: "pb-editor-page-element-advanced-settings-iframe",
+    type: "pb-editor-page-element-advanced-settings",
+    elementType: "iframe",
+    render({ Bind }) {
+        return (
+            <Tab icon={<IFrameIcon />} label="IFrame">
+                <Grid>
+                    <Cell span={12}>
+                        <Bind
+                            // Binding the iframe.url in the 'data' property
+                            name={"iframe.url"}
+                            validators={validation.create("required,url")}
+                        >
+                            <Input
+                                label={"IFrame URL"}
+                                description={"Enter an iFrame URL"}
+                            />
+                        </Bind>
+                    </Cell>
+                </Grid>
+            </Tab>
+        );
+    }
+} as PbEditorPageElementAdvancedSettingsPlugin
+
+```
+
+This code shows the settings dialog and asks for the URL of the iframe as shown in the image below.
+
+![Settings dialog](/img/webiny-apps/page-builder/development/development/plugin-reference/editor/iframe/settings-dialog.png)
+
+> In order to add new fields to your settings dialog, you can add more `PbEditorPageElementAdvancedSettingsPlugin` repeat the same process we've just shown. This will allow you to create other plugins too.
```

Comment:
Check grammar and the structure here. Not sure what was the intention.

![image](https://user-images.githubusercontent.com/5121148/82034365-61523c80-969e-11ea-8f5e-9b3ac835facf.png)

Reply:
I wanted to let people know that they can add more tabs in the settings, like the pagesList.

I checked and will change the text to:

`
 To add more tabs in the settings dialog, you can add more `PbEditorPageElementAdvancedSettingsPlugin` blocks. This will allow you to create other plugins too.
`

What do you think?

Reply:
I would avoid this, just to reduce the complexity. Users already got the most important piece of info here, and that is that they can use the `PbEditorPageElementAdvancedSettingsPlugin` in order to define how the settings dialog will look like.

Reply:
Okay then, maybe that information is a nice to add on a video tutorial. I will remove this part.